components/server/src/workspace/workspace-starter.ts

```diff
@@ -74,9 +76,15 @@ export class WorkspaceStarter {
         const span = TraceContext.startSpan("WorkspaceStarter.launchWindowsWorkspace", ctx);
 
         try {
+
+            var instance = await this.workspaceDb.trace({span}).updateInstancePartial(instanceId, { status: {phase: 'creating', conditions: {}} });
```

Comment:
@daniel-wachira is there a reason why this is moved up ?

Reply:
In the previous setup the workspace status was not set to 'creating' until after we have actually created the workspace and are waiting for confirmation. It seems more correct to have the initialization routine happen while the instance is in creating status - it also makes events a bit clearer when debugging.